src/sentry/models/authprovider.py

```diff
@@ -145,3 +154,30 @@ def disable_scim(self, user):
 
     def get_audit_log_data(self):
         return {"provider": self.provider, "config": self.config}
+
+
+def handle_auth_provider_post_save(instance, **kwargs):
+    from sentry.models import IdentityProvider
+
+    IdentityProvider.objects.create(
+        organization=instance.organization,
+        type=instance.provider,
+        provider=instance.provider,
+        external_id=None,
+        provider_id=None,
+        config=instance.config,
+        is_sso=True,
+        sso_default_team=None,
+        sso_default_role=instance.default_role,
+        sso_default_global_access=instance.default_global_access,
+        sso_flags=instance.flags,
+        authprovider=instance,
+    )
```

Comment:
This will run every time an `AuthProvider` is saved, not just on create. I feel like you want to use `create_or_update` here.